charmcraft/commands/build.py

```diff
@@ -246,19 +247,28 @@ def run(self, bases_indices: Optional[List[int]] = None) -> List[str]:
 
     def pack_charm_in_instance(self, instance: Executor, bases_index: int) -> str:
         """Pack instance in Charm."""
+        charm_name = format_charm_file_name(
+            self.metadata.name, self.config.bases[bases_index]
+        )
+
         try:
             instance.execute_run(
                 ["charmcraft", "pack", "--bases-index", str(bases_index)],
+                check=True,
                 cwd=get_managed_environment_project_path().as_posix(),
             )
         except subprocess.CalledProcessError as error:
+            # Capture log from instance and raise error.
+            instance_log_path = get_managed_environment_log_path()
+            local_log_path = pathlib.Path(self.charmdir, charm_name + ".log")
+            instance.pull_file(source=instance_log_path, destination=local_log_path)
+
             raise CommandError(
-                f"Failed to build charm for bases index '{bases_index}'."
+                f"Failed to build charm for bases index '{bases_index}'. The logs "
+                f"for this charm may be found in {str(local_log_path)!r}."
```

Comment:
I don't like producing TWO log files for the user. One of them polluting the work directory doesn't sound nice either.

What if we get this log from the managed environment, and append its content to the "main log"?

Reply:
I'm open to suggestion as I'm not sure this is ideal either, but I believe the current log implementation would need a bit of work to make that happen cleanly.

On the flip side, this only happens when there is an error and the output log will most likely be analyzed by the user. Having it conveniently located and concise is a benefit.